### ISSUE TITLE
[K-Bug] kprove fails to recognize isKResult as a function

### ISSUE THREAD

**Issue by virgil-serbanuta** [runtimeverification/k#3219]
### What component is the issue in?

Front-End

### Which command

- [ ] kompile
- [ ] kast
- [ ] krun
- [X] kprove
- [ ] kprovex
- [ ] ksearch

### What K Version?

v5.5.132-0-g96ce008122-dirty

### Operating System

Linux

### K Definitions (If Possible)

a.k
```
module A
  imports BOOL

  syntax A ::= "a"
  syntax KResult ::= A

  rule isKResult(a) => true [simplification]
endmodule
```

proof.k
```
module PROOF
  imports A
  claim a => a
endmodule
```

### Steps to Reproduce

```
kompile a.k --backend haskell && kprove proof.k
```

### Expected Results

KProve manages to compile proof.k.

**Comment by radumereuta** [comment]
@ana-pantilie this looks a lot like the problem Nico had.

**Comment by ehildenb** [comment]
We should inspect the Kore first to make sure it's making sense. Potential issues:

- `isKResult` is getting the `function` attribute in the generated Kore?
- If so, then what _could_ be happening is that `isKResult` is getting the `function` attribute _after_ the `simplification` rules are translated to Kore, and so when that rule is being translated to Kore, it sees it as a _semantic_ rule and not a `simplification` equation, and translates it incorrectly.

So basically, it's probably something to do with how `isKResult` is generated, and it may be generated/modified too late.

**Comment by Robertorosmaninho** [comment]
I reproduced it locally and had the following message error:
```
$ kompile a.k --backend haskell && kprove proof.k

[Warning] Compiler: Could not find main syntax module with name A-SYNTAX in
definition.  Use --syntax-module to specify one. Using A as default.
[Error] Compiler: Simplification rules expect function/functional/mlOp symbols
at the top of the left hand side term.
	Source(/Users/robertorosmaninho/rv/tests/issue_3219/a.k)
	Location(7,8,7,28)
	7 |	  rule isKResult(a) => true [simplification]
	  .	       ^~~~~~~~~~~~~~~~~~~~
```

The `symbol` for `isKResult` has the `function` attribute as expected:
```
symbol LblisKResult{}(SortK{}) : SortBool{} [format{}("%cisKResult%r %c(%r %1 %c)%r"), function{}(), functional{}(), left{}(), predicate{}("KResult"), priorities{}(), right{}(), terminals{}("1101"), total{}()]
```

The `axiom` has the `functional`:
```
axiom{R} \exists{R} (Val:SortBool{}, \equals{SortBool{}, R} (Val:SortBool{}, LblsKResult{}(K0:SortK{}))) [functional{}()] // functional
```

The complete `definition.kore` is available [here](https://pastebin.pl/view/697ee8b1).
